tests/io_utils.go

```diff
@@ -85,24 +88,32 @@ func CreateSCSIDisk(nodeName string, opts []string) (address string, device stri
 	_, err := ExecuteCommandInVirtHandlerPod(nodeName, args)
 	ExpectWithOffset(1, err).NotTo(HaveOccurred(), "Failed to create faulty disk")
 
-	args = []string{UsrBinVirtChroot, Mount, Proc1NsMnt, "exec", "--", "/usr/bin/lsscsi"}
-	stdout, err := ExecuteCommandInVirtHandlerPod(nodeName, args)
-	ExpectWithOffset(1, err).NotTo(HaveOccurred(), "Failed to find out address of  SCSI disk")
-
-	// Example output
-	// [2:0:0:0]    cd/dvd  QEMU     QEMU DVD-ROM     2.5+  /dev/sr0
-	// [6:0:0:0]    disk    Linux    scsi_debug       0190  /dev/sda
-	lines := strings.Split(stdout, "\n")
-	for _, line := range lines {
-		if strings.Contains(line, "scsi_debug") {
-			line = strings.TrimSpace(line)
-			disk := strings.Split(line, " ")
-			address = disk[0]
-			address = address[1 : len(address)-1]
-			device = disk[len(disk)-1]
-			break
+	EventuallyWithOffset(1, func() bool {
+		args = []string{UsrBinVirtChroot, Mount, Proc1NsMnt, "exec", "--", "/usr/bin/lsscsi"}
+		stdout, err := ExecuteCommandInVirtHandlerPod(nodeName, args)
+		if err != nil {
+			log.Log.Infof("Error executing lsscsi: %s", err.Error())
+			return false
 		}
-	}
+
+		// Example output
+		// [2:0:0:0]    cd/dvd  QEMU     QEMU DVD-ROM     2.5+  /dev/sr0
+		// [6:0:0:0]    disk    Linux    scsi_debug       0190  /dev/sda
+		lines := strings.Split(stdout, "\n")
+		for _, line := range lines {
+			if strings.Contains(line, "scsi_debug") {
+				line = strings.TrimSpace(line)
+				disk := strings.Split(line, " ")
+				address = disk[0]
+				address = address[1 : len(address)-1]
+				device = disk[len(disk)-1]
+				break
+			}
+		}
+		log.Log.Infof("Device path is: %s", device)
+
+		return filepath.IsAbs(device)
+	}, 20*time.Second, 5*time.Second).Should(BeTrue(), "Can't execute lsscsi or path still not populated")
```

Comment:
It will be better to return an error and assert on its absence.
In the current form, you depend on the logs to understand the problem and that is not nice.

Could you please change this to an error returned?

Reply:
ofc, done